pkg/pillar/cmd/domainmgr/domainmgr.go

```diff
@@ -3473,9 +3479,84 @@ func updateUsbAccess(ctx *domainContext) {
 
 func updateVgaAccess(ctx *domainContext) {
 
-	log.Functionf("updateVgaAccess(%t)", ctx.usbAccess)
-	// TODO: we might need some extra work here for some VGA devices
-	// that do not enable output upon HDMI cable attachment
+	log.Functionf("updateVgaAccess(%t)", ctx.vgaAccess)
+
+	if ctx.vgaAccess {
+		// If VGA is disabled, we need to first bring any VGA PCIe adapter back
+		updatePortAndPciBackIoBundleAll(ctx)
+		checkIoBundleAll(ctx)
+
+		// Nothing to do if VGA is already enabled
+		if vgaSwitch {
+			// VGA access was set to true and it was disabled before, so we
+			// need to perform the "switch VGA back" operations:
+			//
+			// 1. Re-bind framebuffer drivers
+			// 2. Restore activated VTs
+			// 3. Switch back to the last active TTY
+			if err := fbBindAll(); err != nil {
+				log.Errorf("Cannot bind framebuffer drivers: %v", err)
+			}
+			if err := vtBindAll(); err != nil {
+				log.Errorf("Cannot bind Virtual Terminals: %v", err)
+			}
+			if err := chvt(currentTTY); err != nil {
+				log.Errorf("Cannot switch to VT: %v", err)
+			}
+			vgaSwitch = false
+		}
+
+		return
+	} else {
```

Comment:
```suggestion
	} 
```
you return anyways above